dev-utils/visualise-ui-hierarchy/tree2dot

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+import sys
+import fileinput
+from enum import Enum
+
+
+class Node:
+    class Type(Enum):
+        SOURCE = "red"
+        POLY_LOOK = "green"
+        THIRD_PARTY = "blue"
+
+    def __init__(self, key, value, type):
+        self.key = key
+        self.value = value
+        self.type = type
+
+
+class Graph:
+    def __init__(self):
+        self.__nodes = {}
+        self.edges = set()
+
+    def register_node(self, value, type):
+        if value not in self.__nodes:
+            key = str(len(self.__nodes) + 1)
+            self.__nodes[value] = Node(key, value, type)
```

Comment:
A bit confusing to index nodes by value, not by keys...

Reply:
That's just a lookup map, we do lookups by value...

Reply:
Right, but as long as you've got key and value, using value to index is confusing. It's a naming nit, I guess. Maybe use `id` instead of `key`? And `name` instead of `value`?

Reply:
Got it, I'll see what Graphviz actually calls what I call "key" here, I think it actually calls it "ID", and "value" can be either "path" (what it is) or "label" (what it becomes in the dot file)

Reply:
Went for "ID" and "label" now, since that's what those are called in Graphviz. Should be less confusing now!